js/modules/k6/k6.go

```diff
@@ -31,22 +30,64 @@ import (
 	"github.com/dop251/goja"
 
 	"go.k6.io/k6/js/common"
-	"go.k6.io/k6/lib"
+	"go.k6.io/k6/js/modules"
 	"go.k6.io/k6/stats"
 )
 
-// K6 is just the module struct.
-type K6 struct{}
+var (
+	// ErrGroupInInitContext is returned when group() are using in the init context.
+	ErrGroupInInitContext = common.NewInitContextError("Using group() in the init context is not supported")
 
-// ErrGroupInInitContext is returned when group() are using in the init context.
-var ErrGroupInInitContext = common.NewInitContextError("Using group() in the init context is not supported")
+	// ErrCheckInInitContext is returned when check() are using in the init context.
+	ErrCheckInInitContext = common.NewInitContextError("Using check() in the init context is not supported")
+)
+
+type (
+	// RootModule is the global module instance that will create module
+	// instances for each VU.
+	RootModule struct{}
+
+	// K6 represents an instance of the k6 module.
+	K6 struct {
+		vu  modules.VU
+		obj *goja.Object
+	}
+)
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &K6{}
+)
+
+// New returns a pointer to a new RootModule instance.
+func New() *RootModule {
+	return &RootModule{}
+}
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (*RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	rt := vu.Runtime()
+	mi := &K6{vu: vu, obj: rt.NewObject()}
+
+	mustExport := func(name string, value interface{}) {
+		if err := mi.obj.Set(name, value); err != nil {
+			common.Throw(rt, err)
+		}
+	}
+
+	mustExport("check", mi.Check)
+	mustExport("fail", mi.Fail)
+	mustExport("group", mi.Group)
+	mustExport("randomSeed", mi.RandomSeed)
+	mustExport("sleep", mi.Sleep)
 
-// ErrCheckInInitContext is returned when check() are using in the init context.
-var ErrCheckInInitContext = common.NewInitContextError("Using check() in the init context is not supported")
+	return mi
+}
 
-// New returns a new module Struct.
-func New() *K6 {
-	return &K6{}
+// Exports returns the exports of the k6 module.
+func (mi *K6) Exports() modules.Exports {
+	return modules.Exports{Default: mi.obj}
```

Comment:
I would argue this should set only `Named` and then you can also drop the `obj` and have the map inlined here similar to 
https://github.com/grafana/k6/blob/3a604bf408ec278f71da2f577c76edc7778aece2/js/modules/k6/metrics/metrics.go#L162-L171

After all I would expect that users will use that as `import { sleep } from "k6";sleep()` not as `import k6 from "k6"; k6.sleep()`